core/src/data/interval/primitive.rs

```diff
@@ -109,6 +128,17 @@ impl Div<i8> for Interval {
     }
 }
 
+impl Div<i16> for Interval {
+    type Output = Self;
+
+    fn div(self, rhs: i16) -> Self {
+        match self {
+            Interval::Month(v) => Interval::Month(v / rhs as i32),
+            Interval::Microsecond(v) => Interval::Microsecond(v / rhs as i64),
+        }
```

Comment:
This `match self` part appears similarly on every `impl Div<_> for Interval { ... }`s.
Would be nice to refactor them with a helper function in near future.